File: common/src/androidMain/kotlin/com/blockstream/common/devices/NfcCardChannel.java
//package com.blockstream.common.devices;
//
//import android.nfc.tech.IsoDep;
//import android.util.Log;
////import org.satochip.io.APDUCommand;
////import org.satochip.io.APDUResponse;
////import org.satochip.io.CardChannel;
//
//import java.io.IOException;
//
///**
// * Implementation of the CardChannel interface using the Android NFC API.
// */
//public class NfcCardChannel implements CardChannel {
//  private static final String TAG = "CardChannel";
//
//  private IsoDep isoDep;
//
//  public NfcCardChannel(IsoDep isoDep) {
//    this.isoDep = isoDep;
//  }
//
//  @Override
//  public APDUResponse send(APDUCommand cmd) throws IOException {
//    byte[] apdu = cmd.serialize();
//    Log.d(TAG, String.format("COMMAND CLA: %02X INS: %02X P1: %02X P2: %02X LC: %02X", cmd.getCla(), cmd.getIns(), cmd.getP1(), cmd.getP2(), cmd.getData().length));
//    byte[] resp = this.isoDep.transceive(apdu);
//    APDUResponse response = new APDUResponse(resp);
//    Log.d(TAG, String.format("RESPONSE LEN: %02X, SW: %04X %n-----------------------", response.getData().length, response.getSw()));
//    return response;
//  }
//
//  @Override
//  public boolean isConnected() {
//    return this.isoDep.isConnected();
//  }
//}

File: common/src/androidMain/kotlin/com/blockstream/common/devices/NfcCardChannel.kt
package com.blockstream.common.devices

import android.nfc.tech.IsoDep
import android.util.Log

/**
 * Implementation of the CardChannel interface using the Android NFC API.
 */
class NfcCardChannel(private val isoDep: IsoDep) : CardChannel {
    companion object {
        private const val TAG = "CardChannel"
    }

    override fun send(cmd: ApduCommand): ApduResponse {
        val apdu = cmd.serialize()
        Log.d(TAG, "COMMAND CLA: %02X INS: %02X P1: %02X P2: %02X LC: %02X".format(
            cmd.cla, cmd.ins, cmd.p1, cmd.p2, cmd.data.size
        ))

        val resp = isoDep.transceive(apdu)
        val response = ApduResponse(resp)
        Log.d(TAG, "RESPONSE LEN: %02X, SW: %04X %n-----------------------".format(
            response.getData().size, response.getSw()
        ))

        return response
    }

    override fun isConnected(): Boolean {
        return isoDep.isConnected
    }
}